src/config.rs

```diff
@@ -6,70 +6,100 @@ use std::{
 use serde::Deserialize;
 use tracing::debug;
 
+/// Configuration for service needed for
+/// orchestrator to communicate with it
 #[derive(Debug, Clone, Deserialize)]
 pub struct ServiceConfig {
     pub hostname: String,
     pub port: Option<u16>,
     pub tls: Option<Tls>,
 }
 
+/// TLS provider
 #[derive(Debug, Clone, Deserialize)]
 #[serde(untagged)]
 pub enum Tls {
     Name(String),
     Config(TlsConfig),
 }
 
+/// Client TLS configuration
 #[derive(Debug, Clone, Deserialize)]
 pub struct TlsConfig {
     pub cert_path: Option<PathBuf>,
     pub key_path: Option<PathBuf>,
     pub client_ca_cert_path: Option<PathBuf>,
 }
 
+/// Generation service provider
 #[derive(Debug, Clone, Copy, Deserialize)]
 #[serde(rename_all = "lowercase")]
 pub enum GenerationProvider {
     Tgis,
     Nlp,
 }
 
+/// Generate service configuration
 #[derive(Debug, Clone, Deserialize)]
 pub struct GenerationConfig {
+    /// Generation service provider
     pub provider: GenerationProvider,
+    /// Generation service connection information
     pub service: ServiceConfig,
 }
 
+/// Chunker parser type
 #[derive(Debug, Clone, Copy, Deserialize)]
 #[serde(rename_all = "lowercase")]
 pub enum ChunkerType {
     Sentence,
     All,
 }
 
+/// Configuration for each chunker
 #[allow(dead_code)]
 #[derive(Debug, Clone, Deserialize)]
 pub struct ChunkerConfig {
+    /// Chunker type
     pub r#type: ChunkerType,
+    /// Chunker service connection information
     pub service: ServiceConfig,
 }
 
+/// Configuration parameters applicable to each detector
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+pub struct DetectorConfigParams {
+    /// Default threshold with which to filter detector results by score
+    pub default_threshold: Option<f32>,
+}
+
+/// Configuration for each detector
 #[derive(Debug, Clone, Deserialize)]
 pub struct DetectorConfig {
+    /// Detector service connection information
     pub service: ServiceConfig,
+    /// ID of chunker that this detector will use
     pub chunker_id: String,
-    //pub config: HashMap<String, String>,
+    /// Optional detector configuration parameters
+    pub config: Option<DetectorConfigParams>,
```

Comment:
Can we drop `config: Option<DetectorConfigParams>` and add `default_threshold` to `DetectorConfig`? The nested config object seems unnecessary.

I also think we should make it `f32` instead of `Option<f32>` since its a default.